thrust/thrust/iterator/detail/iterator_category_to_system.h

```diff
@@ -38,14 +38,23 @@ _CCCL_HOST_DEVICE auto cat_to_system_impl(...) -> void;
 
 _CCCL_HOST_DEVICE auto cat_to_system_impl(const input_host_iterator_tag&) -> host_system_tag;
 _CCCL_HOST_DEVICE auto cat_to_system_impl(const output_host_iterator_tag&) -> host_system_tag;
+_CCCL_HOST_DEVICE auto cat_to_system_impl(const forward_host_iterator_tag&) -> host_system_tag;
+_CCCL_HOST_DEVICE auto cat_to_system_impl(const bidirectional_host_iterator_tag&) -> host_system_tag;
+_CCCL_HOST_DEVICE auto cat_to_system_impl(const random_access_host_iterator_tag&) -> host_system_tag;
 
 _CCCL_HOST_DEVICE auto cat_to_system_impl(const input_device_iterator_tag&) -> device_system_tag;
 _CCCL_HOST_DEVICE auto cat_to_system_impl(const output_device_iterator_tag&) -> device_system_tag;
+_CCCL_HOST_DEVICE auto cat_to_system_impl(const forward_device_iterator_tag&) -> device_system_tag;
+_CCCL_HOST_DEVICE auto cat_to_system_impl(const bidirectional_device_iterator_tag&) -> device_system_tag;
+_CCCL_HOST_DEVICE auto cat_to_system_impl(const random_access_device_iterator_tag&) -> device_system_tag;
 
 template <typename Category>
 struct iterator_category_to_system
 {
   using type = decltype(cat_to_system_impl(Category{}));
 };
+
+template <typename Category>
+using iterator_category_to_system_t = typename iterator_category_to_system<Category>::type;
```

Comment:
I believe this should just be
```suggestion
using iterator_category_to_system_t = decltype(cat_to_system_impl(Category{}));
```